src/scripts/main.js

```diff
@@ -1 +1,75 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', (e) => {
+    if (e.button === 0) {
+      resolve('First promise was resolved');
+    }
+  });
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
```

Comment:
The first promise will never reject if the user clicks the left mouse button before the timeout. Consider clearing the timeout when the promise is resolved to prevent the rejection from occurring.